src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp

```diff
@@ -937,87 +1174,348 @@ void BrgemmEmitter::emit_brgemm_kernel_call(const brgemm_kernel_t *brgKernel, in
+            transposed_shape[i] = original_shape[layout[i]];
+        }
+        // The idea here is to find "2" (for 4D shapes) in the layout and multiply dimensions that are to the right
+        // This implies that "3" is the last layout value, otherwise this layout is not supported.
+        // counting from the end since shape could be prepended with ones
+        const int64_t num_last_dims = layout.end() - std::find(layout.begin(), layout.end(), layout.size() - 2) - 1;
+        if (layout.back() != layout.size() - 1 || num_last_dims < 1)
+            IE_THROW() << "BrgemmRepackEmitter detected invalid layout values: " <<
+                       "check that this shape + layout combination is schedulable";
+        leading_dimension = std::accumulate(original_shape.end() - num_last_dims, original_shape.end(), 1, std::multiplies<size_t>());
+    }
+
+    N = *(transposed_shape.rbegin());
+    K = *(transposed_shape.rbegin() + 1);
+
+    const bool isAMXSupported = mayiuse(avx512_core_bf16_amx_int8) || mayiuse(avx512_core_bf16_amx_bf16);
+    const auto use_amx = isAMXSupported && brgemm_prc_in0 != ov::element::f32 && (K % brgemmVNNIFactor == 0) && (N % brgemmVNNIFactor == 0);
+
+    N_blk = brgemm_prc_in1 == ov::element::f32 ? N :
+            brgemm_prc_in1 == ov::element::bf16 ? 32 : 64;
+    K_blk = use_amx ? brgemm_prc_in0 == ov::element::bf16 ? 32 : 64
+                    : K;
+    N_tail = N % N_blk;
+    K_tail = K % K_blk;
+    LDB = brgemm_prc_in1 == ov::element::f32 ? leading_dimension : rnd_up(N, N_blk);
+
+    const auto dt_in0 = static_cast<dnnl_data_type_t>(DnnlExtensionUtils::IEPrecisionToDataType(InferenceEngine::details::convertPrecision(brgemm_prc_in0)));
+    const auto dt_in1 = static_cast<dnnl_data_type_t>(DnnlExtensionUtils::IEPrecisionToDataType(InferenceEngine::details::convertPrecision(brgemm_prc_in1)));
+    init_brgemm_copy(kernel, leading_dimension, N_blk, N_tail, LDB, K - K_tail, use_amx, dt_in0, dt_in1);
+
+    in_offset = brgemm_repack->get_offset_in();
+    out_offset = brgemm_repack->get_offset_out();
+}
+
+void BrgemmCopyBBaseEmitter::init_brgemm_copy(std::unique_ptr<matmul::jit_brgemm_matmul_copy_b_t>& kernel,
+                                              size_t N, size_t N_blk, size_t N_tail, size_t LDB, size_t K,
+                                              bool is_with_amx, dnnl_data_type_t dt_in0, dnnl_data_type_t dt_in1) const {
+    matmul::brgemm_matmul_conf_t brgCopyKernelConf;
+    brgCopyKernelConf.src_dt = dt_in0;
+    brgCopyKernelConf.wei_dt = dt_in1;
+    brgCopyKernelConf.wei_n_blk = N_blk;
+    brgCopyKernelConf.wei_tag = dnnl_abcd;  // What's about other ranks?
+    brgCopyKernelConf.copy_B_wei_stride = 0;
+    brgCopyKernelConf.LDB = LDB;
+    brgCopyKernelConf.N = N;
+    brgCopyKernelConf.N_tail = N_tail;
+    brgCopyKernelConf.N_blk = N_blk;
+    brgCopyKernelConf.K = K;
+    brgCopyKernelConf.K_blk = K;
+    brgCopyKernelConf.N_chunk_elems = brgCopyKernelConf.N_blk;
```

Comment:
Minor: 
1) To avoid dealing with code-analyzer reports later, it would be nice to eliminate implicit signed-unsigned conversions here (L1241, 12,44-12499)
2) The function parameter have the same names as class fields.